src/app/models/vault.ts

```diff
@@ -0,0 +1,32 @@
+import {Contract} from './contract';
+import {IContract} from './icontract';
+import {map} from 'rxjs/operators';
+
+export class Vault implements IContract {
+    id: number;
+    name: string;
+    underlyingUnit: number;
+    updatedBlock: number;
+    contract: Contract;
+    controller: Contract;
+    decimals: number;
+    governance: Contract;
+    strategy: Contract;
+    underlying: Contract;
+    symbol: string;
+    isCRV() {
+        return this.underlying?.type === 4 && this.contract?.name?.match(/CRV/);
+    }
+    isOtherLP() {
+        return this.underlying?.type === 2 && !this.underlying?.name?.match(/^UNI_LP/);
+    }
+    isStableCoin() {
```

Comment:
isStablecoin need for simplifying price calculation
it was an old shortcut, I guess we can remove this logic
anyway, looks like you use the wrong logic
stableCoin list is manually filled, I'm not sure that we can detect it